### DISCUSSION TITLE
Can I exclude a route from Next.JS entirely?

### DISCUSSION THREAD

**Discussion by sysRoot** [vercel/next.js#29261] [Help] [answered]
Theoretically, I'd like to use HashOver comment system, which is a php app. I am using Caddy as a reverse proxy. I am NOT asking for help with Caddy configuration, or how to setup my comments

What I'd like to do is EXCLUDE a path from being handled by NextJS so that the directive kicks in from Caddy.

```
example.com {
  reverse_proxy localhost:3000  {
    header_down Strict-Transport-Security max-age=31536000;
   }
  php_fastcgi {
    root /mpcom/* /opt/mpcom
    index off
   }
}
```

As such, you can see, which I may have to further tinker with the setup for the PHP part... I'd like to exclude NextJS from handling, and by logic: serving a 404, for the `example.com/mpcom` url.

Is this possible with NextJS? I'm comfortable with modifying config files.

**Comment by sysRoot** [comment, marked as answer]
It looks like this might be what I want to do here:

```
// server.js
const { createServer } = require('http')
const { parse } = require('url')
const next = require('next')

const dev = process.env.NODE_ENV !== 'production'
const app = next({ dev })
const handle = app.getRequestHandler()

app.prepare().then(() => {
  createServer((req, res) => {
    // Be sure to pass `true` as the second argument to `url.parse`.
    // This tells it to parse the query portion of the URL.
    const parsedUrl = parse(req.url, true)
    const { pathname, query } = parsedUrl

    if (pathname === '/mpcom') {
      handle(req, res, parsedUrl)
    } else {
      app.render(req, res, '/', query)
    }

  }).listen(3000, (err) => {
    if (err) throw err
    console.log('> Ready on http://localhost:3000')
  })
})

```

I'm going to test and report back, took me a bit to find that page in relation to the google queries I was using

**Reply by sysRoot** [reply]
This self-suggestion did not function as expected, perhaps I lack true understanding of what's happening.

**Reply by wwwdepot** [reply]
All you need to do is configure `Caddyfile`, Caddy will send all traffic with path starts with `/mpcom` to your php app & proxy all others to your Next.js app.

**Reply by sysRoot** [reply]
Thank you :)
